### PR TITLE
Annihilate reddit cog

### LINKED ISSUE #1301: Annihilate the `!reddit` command
Once the `#reddit` channel will be managed by Sir-Lancebot (see https://github.com/python-discord/sir-lancebot/issues/530), the [Reddit command](https://github.com/python-discord/bot/blob/master/bot/exts/info/reddit.py#L257-L260) should be removed from Python. 

This is stalled until https://github.com/python-discord/sir-lancebot/issues/530 is completed.

### PR DESCRIPTION
**Pull request by RohanJnr**
closes #1301 

Remove the Reddit cog and traces of reddit constants.
The reddit cog is moving to sir-lancebot.

This PR should be merged along with https://github.com/python-discord/sir-lancebot/pull/569

### CODE REVIEW

File: config-default.yml [hunk 1]
```diff
@@ -512,7 +503,6 @@ duck_pond:
         - *PYNEWS_CHANNEL
         - *PYEVENTS_CHANNEL
         - *MAILING_LISTS
-        - *REDDIT_CHANNEL
```
Comment (kwzrd) [inline config-default.yml:515]: Does removing this make sense? The channel will continue to exist, right? This is config for a different feature.

**Review by kwzrd — COMMENTED** [review]
I pointed out one place in the config that I'm not sure whether it should be removed.

On the other hand, [these emojis](https://github.com/python-discord/bot/blob/2e456588fdcdf6286636a86d4083f43b62b1f995/config-default.yml#L73-L76) seem unused now.

File: config-default.yml (hunk 1 above)
Comment (RohanJnr) [reply config-default.yml:515]: we could keep them 👍

**Review by MarkKoz — APPROVED** [review]

File: bot/utils/time.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def parse_rfc1123(stamp: str) -> datetime.datetime:
     return datetime.datetime.strptime(stamp, RFC1123_FORMAT).replace(tzinfo=datetime.timezone.utc)
 
 
-# Hey, this could actually be used in the off_topic_names and reddit cogs :)
+# Hey, this could actually be used in the off_topic_names cog :)
```
Comment (Xithrius) [inline bot/utils/time.py:147]:
```suggestion
```

**Review by Xithrius — APPROVED** [review]

**Pull request merged**